### PR TITLE
[RHBOP] Nightly sources generation fix

### PR DESCRIPTION
**Pull request by lampajr**
### JIRA

- https://issues.redhat.com/browse/BXMSPROD-1826

### Referenced pull requests
- https://github.com/kiegroup/optaplanner-quickstarts/pull/421
- https://github.com/jboss-integration/rhbop-optaplanner/pull/1
- https://github.com/kiegroup/optaplanner/pull/2214
- https://github.com/kiegroup/optaplanner/pull/2223

Changes:
- Rename optaweb sources folder name 
- Remove the following quickstarts examples from sources generation:
  - technology
    - python
    - kubernetes
    
### Checklist
- [ ] Documentation updated if applicable.
- [ ] Release notes updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] tests</b>
  
- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: 
    please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: 
    add the label `run_fdb`

- for a <b>compile downstream build</b>  
  please add comment: <b>Jenkins run cdb</b>

- for a <b>full production downstream build</b>  
  please add comment: <b>Jenkins execute product fdb</b>

- for an <b>upstream build</b>  
  please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] mandrel</b>
</details>

### CI Status

 You can check OptaPlanner repositories CI status from [Chain Status webpage](https://kiegroup.github.io/optaplanner/).

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - optaplanner-quickstarts job** `#442` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/optaplanner/job/main/job/pullrequest/job/optaplanner.tests.downstream.optaplanner-quickstarts/442/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-10-03T13:03:21.460Z] 0	/home/jenkins/workspace/KIE/kogito/1.13.x<br/>[2022-10-03T13:03:21.460Z] 0	/home/jenkins/workspace/KIE/kogito/1.27.x<br/>[2022-10-03T13:03:21.460Z] 4.0K	/home/jenkins/workspace/KIE/kogito<br/>[2022-10-03T13:03:21.460Z] du: cannot access ‘/home/jenkins/workspace/KIE/main/deployedRep/kie-wb-common_ws-cleanup_1664802197224/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/target/apidocs/org/kie/workbench/common/stunner/core/client/animation/Animation.AnimationCallback.html’: No such file or directory<br/>[2022-10-03T13:03:21.460Z] du: cannot access ‘/home/jenkins/workspace/KIE/main/deployedRep/kie-wb-common_ws-cleanup_1664802197224/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/target/apidocs/org/kie/workbench/common/stunner/core/client/animation/AnimationFactory.html’: No such file or directory<br/>[2022-10-03T13:03:21.460Z] du: cannot access ‘/home/jenkins/workspace/KIE/main/deployedRep/kie-wb-common_ws-cleanup_1664802197224/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/target/apidocs/org/kie/workbench/common/stunner/core/client/animation/AnimationHandle.html’: No such file or directory<br/>[2022-10-03T13:03:21.460Z] du: cannot access ‘/home/jenkins/workspace/KIE/main/deployedRep/kie-wb-common_ws-cleanup_1664802197224/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/target/apidocs/org/kie/workbench/common/stunner/core/client/animation/package-summary.html’: No such file or directory<br/>[2022-10-03T13:03:21.460Z] du: cannot access ‘/home/jenkins/workspace/KIE/main/deployedRep/kie-wb-common_ws-cleanup_1664802197224/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/target/apidocs/org/kie/workbench/common/stunner/core/client/animation/package-tree.html’: No such file or directory<br/>[2022-10-03T13:03:21.460Z] 1.2G	/home/jenkins/workspace/KIE/main/deployedRep<br/>[2022-10-03T13:03:21.460Z] 1.2G	/home/jenkins/workspace/KIE/main<br/>[2022-10-03T13:03:21.460Z] 3.0M	/home/jenkins/workspace/KIE/optaplanner/main<br/>[2022-10-03T13:03:21.460Z] 3.0M	/home/jenkins/workspace/KIE/optaplanner<br/>[2022-10-03T13:03:21.460Z] 1.2G	/home/jenkins/workspace/KIE<br/>[2022-10-03T13:03:21.460Z] 1.2G	/home/jenkins/workspace<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>Stage "Install build-chain tool" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-10-03T13:03:22.346Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/optaplanner/job/main/job/pullrequest/job/optaplanner.tests.downstream.optaplanner-quickstarts/442/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Review by dupliaka — APPROVED** [review]
LGTM

File: .ci/jenkins/Jenkinsfile.prod.nightly [hunk 1]
```diff
@@ -87,17 +87,17 @@ pipeline {
...
+                            sh "zip -r $optaplannerSourcesFilename $optaplannerSourceDir -x '$optaplannerSourceDir/.git/*' -x '$optaplannerSourceDir/optaplanner-operator/*' -x '$optaplannerSourceDir/target/*'"
                             uploadSources(optaplannerSourcesFilename, folder)
 
                             def quickstartsSourcesFilename = computeSourcesFilename("optaplanner-quickstarts", PME_BUILD_VARIABLES['datetimeSuffix'])
-                            sh "zip -r $quickstartsSourcesFilename $quickstartsSourceDir $optawebSourceDir -x '$quickstartsSourceDir/.git/*' -x '$optawebSourceDir/.git/*' -x '$quickstartsSourceDir/target/*' -x '$optawebSourceDir/target/*'"
+                            sh "zip -r $quickstartsSourcesFilename $quickstartsSourceDir $optawebSourceDir -x '$quickstartsSourceDir/.git/*' -x '$optawebSourceDir/.git/*' -x '$quickstartsSourceDir/target/*' -x '$optawebSourceDir/target/*' -x '$quickstartsSourceDir/technology/python/*' -x '$quickstartsSourceDir/technology/kubernetes/*'"
```
Comment (mareknovotny) [inline .ci/jenkins/Jenkinsfile.prod.nightly:100]: maybe if you use https://github.com/jboss-integration/rhbop-optaplanner you don't need to zip and exclude not wanted directories ?

Comment (lampajr) [reply .ci/jenkins/Jenkinsfile.prod.nightly:100]: Yeah, I think that using rhbop-optaplanner we can completely remove the zip part and directly upload the generated zip artifacts.

**Comment by lampajr** [comment]
Jenkins retest this

**Review by mareknovotny — APPROVED** [review]

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**